dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1568,52 +1576,37 @@ protected void unsetCacheServletWrapperFactory(CacheServletWrapperFactory factor
         // no-op intended here to avoid cacheServletWrapperFactory being null when switching service implementations
     }
 
-    
-    @Reference(service=ServletVersion.class, cardinality=ReferenceCardinality.MANDATORY, policy=ReferencePolicy.DYNAMIC, policyOption=ReferencePolicyOption.GREEDY)
-    protected synchronized void setVersion(ServiceReference<ServletVersion> reference) {
-        String methodName = "setVersion";
-        versionRef = reference;
-        WebContainer.loadedContainerSpecLevel = (Integer) reference.getProperty("version");
-        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.debug(tc, methodName, "loadedContainerSpecLevel [ " + WebContainer.loadedContainerSpecLevel + " ]");
-        }
-    }
 
-    protected synchronized void unsetVersion(ServiceReference<ServletVersion> reference) {
-        if (reference == this.versionRef) {
-            versionRef = null;
-            WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
-        }
-    }
-    
-    public static final int SPEC_LEVEL_UNLOADED = -1;
-    public static final int SPEC_LEVEL_30 = 30;
-    public static final int SPEC_LEVEL_31 = 31;
-    public static final int SPEC_LEVEL_40 = 40;
-    public static final int SPEC_LEVEL_50 = 50;
-    private static final int DEFAULT_SPEC_LEVEL = 30;
-    
-    private static int loadedContainerSpecLevel = SPEC_LEVEL_UNLOADED;
-    
-    public static int getServletContainerSpecLevel() {
-        String methodName = "getServletContainerSpecLevel";
+    private static synchronized int loadServletVersion(){
```

Comment:
I see no reason for this to be synchronized

Reply:
Neither can I. Removing